### ISSUE TITLE
Blank chart displayed when `data_server` enabled in docker

### ISSUE THREAD

**Issue by xujiboy** [altair-viz/altair#1787]
Hi,
I'm having trouble to properly display charts on jupyter notebook when it is served from a docker container. The chart can display correctly with "json" `data_transformer` enabled, but not with "data_server" `data_transformer` enabled. Below is a screenshot:
![Screen Shot 2019-11-13 at 9 37 58 PM](https://user-images.githubusercontent.com/10766549/68829559-fcba1480-065d-11ea-93b1-169e4b2eb61d.png)
- the adblocker is disabled on my Chrome
- the javascript console shows 
![Screen Shot 2019-11-13 at 9 40 37 PM](https://user-images.githubusercontent.com/10766549/68829662-46a2fa80-065e-11ea-9f17-d919d3012a36.png)
- my `Dockerfile` looks like this
```Dockerfile
FROM python:3.7.4-slim

WORKDIR /app

COPY ./requirement.txt /app/

RUN pip --no-cache-dir install -r requirement.txt

EXPOSE 8888

VOLUME /app
```
Does this mean the "data_server" connection is somehow blocked in the container? How to correct this situation?

**Comment by jakevdp** [comment]
You may have to use ``data_server_proxied`` instead, since a container essentially acts as a remote system hosted on your machine. See https://github.com/altair-viz/altair_data_server/#remote-systems for more information.

**Comment by xujiboy** [comment]
Hi @jakevdp thank you very much for the suggestion. I have pip installed `jupyter-server-proxy` and rebuilt the container, but still can not see the chart:
![Screen Shot 2019-11-14 at 10 19 19 AM](https://user-images.githubusercontent.com/10766549/68884718-99190100-06c8-11ea-9bb8-67a2161b9363.png)
It seems that the browser is trying to fetch the served json file ("http://localhost:8889/notebooks/proxy/16786/024361a3-42d5-43f8-a3dd-e06513fb3dcf.json") that can not be found. Is there some special configuration that needs to be done in the Dockerfile?

**Comment by jakevdp** [comment]
If that filepath cannot be resolved, I suspect that the jupyter-server-proxy is not installed on your frontend. Indeed, it's not listed in the "Loading extension" console lines in your screenshot.

**Comment by xujiboy** [comment]
I am a bit confused ... the `jupyter-server-proxy` is clearly installed as it shows up in the `pip freeze` output. Is there another way to make sure that `jupyter-server-proxy` is installed at the frontend (jupyter notebook)?

**Comment by jakevdp** [comment]
Yeah, Jupyter frontend extensions are super confusing. You need to make certain it's installed in the Python environment from which you *launch* Jupyter (not the one from which you're running the code *in* the notebook) And you also need to ensure the frontend extension is enabled - see https://github.com/jupyterhub/jupyter-server-proxy for more information.
